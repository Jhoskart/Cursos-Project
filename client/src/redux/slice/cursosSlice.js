import { createSlice } from '@reduxjs/toolkit';

export const cursosSlice = createSlice({
    name: 'cursos',
    initialState: {
        cursos: []
    },
    reducers: {
        getCursos: (state, {payload}) => {
            state.cursos = payload;
        }
    }
});

export const { getCursos } = cursosSlice.actions;