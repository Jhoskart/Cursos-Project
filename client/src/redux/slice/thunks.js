import axios from 'axios';
import { getCursos } from './cursosSlice';
import Swal from 'sweetalert2';

const api = process.env.REACT_APP_API;

export const getCursosRequest = () => async dispatch => {
    try {
        const response = await axios.get(`${api}/cursos`);
        const data = await response.data;	
        dispatch(getCursos(data));
    } catch (error) {
        console.log(error);
    }
}

export const updateCursoRequest = (id, curso) => {
    return async () => {
        try {
            await axios.put(`${api}/cursos/${id}`, curso);
            Swal.fire({
                title: 'Curso actualizado',
                text: 'El curso se actualiz√≥ correctamente',
                icon: 'success',
                showConfirmButton: false,
                footer: '<a class="btn btn-primary" href="/">OK</a>'
            });
        } catch (error) {
            console.log(error);
        }
    }
}


export const postCursoRequest = (curso) => {
    return async () => {
        try {
            const response = await axios.post(`${api}/cursos`,{
            method: 'POST', 
            body: curso
        });	
            
            Swal.fire({
                title: 'Curso creado',
                text: 'El curso se ha creado correctamente',
                icon: 'success',
                showConfirmButton: false,
                footer: '<a class="btn btn-primary" href="/">OK</a>'
            });
            console.log(response);
        } catch (error) {
            Swal.fire({
                title: 'Error',
                text: `${error.response.data.message}`,
                icon: 'error',
                confirmButtonText: 'Ok'
            });
            console.log(error);
        }
    }
}

export const deleteCursoRequest = (id) => async dispatch => {
    try {
        const response = await axios.delete(`${api}/cursos/${id}`);
        Swal.fire({	
            title: 'Curso eliminado',
            text: 'El curso se ha eliminado correctamente',
            icon: 'success',
            showConfirmButton: false,
            footer: '<a class="btn btn-primary" href="/">OK</a>'
        });
        console.log(response);
    } catch (error) {
        Swal.fire({
            title: 'Error',
            text: `${error.response.data.message}`,
            icon: 'error',
            confirmButtonText: 'Ok'
        });
        console.log(error);
    }
}
