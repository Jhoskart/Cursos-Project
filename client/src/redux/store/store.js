import { configureStore } from '@reduxjs/toolkit';
import { cursosSlice } from '../slice/cursosSlice';

export default configureStore({
    reducer: {
        cursos: cursosSlice.reducer
    },
});
