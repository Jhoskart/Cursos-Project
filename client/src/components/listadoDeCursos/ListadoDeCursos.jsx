import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getCursosRequest } from '../../redux/slice/thunks'
import CursoCard from '../cursoCard/CursoCard'
import CreateCurso from '../createCurso/CreateCurso'
import './listadoDeCursos.css'

export default function ListadoDeCursos() {
    const dispatch = useDispatch()
    const cursos = useSelector(state => state.cursos.cursos)
   
    useEffect(() => {
        dispatch(getCursosRequest())
    } , [dispatch])

    return (
        <div className="margin3">
            <h1>Listado de CursosðŸ‘½</h1>
            <div className="content">
                {cursos?.map(curso => (
                    <>
                        <CursoCard 
                            key={curso.id} 
                            name={curso.name} 
                            image={curso.image} 
                            description={curso.description}
                            id={curso.id}
                        />
                    </>
                ))}
            </div>
            <CreateCurso />  
        </div>
    )
}
