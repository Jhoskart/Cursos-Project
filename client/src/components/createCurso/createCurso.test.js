import {screen, render, fireEvent} from '@testing-library/react';
import { Provider } from 'react-redux';
import CreateConnected from './CreateCurso.jsx'; 
import ListadoDeCursos from '../listadoDeCursos/ListadoDeCursos'
import store  from '../../redux/store/store';
import { configure, shallow, mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import configureStore from 'redux-mock-store';
import { getCursos } from '../../redux/slice/cursosSlice';
import axios from 'axios';
axios.defaults.adapter = require('axios/lib/adapters/http');

configure({ adapter: new Adapter() });

describe('CreateCursos', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(
            <Provider store={store}>
                <CreateConnected />
            </Provider>
        );
    })

    it('should render correctly', () => {
        const { container } = render(
            <Provider store={store}>
                <CreateConnected />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    })

    it(' should render a button to create a curso', () => {
        const { container } = render(
            <Provider store={store}>
                <CreateConnected />
            </Provider>
        );
        expect(screen.getByText('Crear Curso')).toBeInTheDocument();
    })

    it('should render a Modal to create a curso and have a form', () => {
        const { container } = render(
            <Provider store={store}>
                <CreateConnected />
            </Provider>
        );
        
        fireEvent.click(screen.getByText('Crear Curso'));
        expect(screen.getByRole('dialog')).toBeInTheDocument(); 
        expect(screen.getByPlaceholderText('Nombre del curso')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('Descripcion del curso')).toBeInTheDocument();
        expect(screen.getByPlaceholderText('Imagen del curso')).toBeInTheDocument();
    })

});
