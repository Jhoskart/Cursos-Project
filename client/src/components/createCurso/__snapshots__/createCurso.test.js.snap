// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCursos should render correctly 1`] = `
<div>
  <div
    class="margin2"
  >
    <button
      class="tamaÃ±o btn btn-primary"
      type="button"
    >
      Crear Curso
    </button>
  </div>
</div>
`;
